### PR TITLE
Update service hooks with useDeepMemo hook

### PR DESCRIPTION
**Pull request by myty**
Fixes: #153 

feat: add deep compare and replace methods
feat: add useDeepMemo hook (uses deepCompare and deepReplace methods)
fix: update `useListService` and `useGetService` to utilize `useDeepMemo` hook

-   [x] Related GitHub issue(s) linked in PR description
-   [x] Destination branch merged, built and tested with your changes
-   [x] Code formatted and follows best practices and patterns
-   [x] Code builds cleanly (no _additional_ warnings or errors)
-   [x] Manually tested
-   [x] Automated tests are passing
-   [x] No _decreases_ in automated test coverage
-   [x] Documentation updated (readme, docs, comments, etc.)
-   [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

File: packages/javascript-react/src/utilities/object-utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+            // Act
+            const result = deepReplace(current, value);
+
+            // Assert
+            expect(result).not.toEqual(current);
+            expect(result.b.d).toEqual(value.b.d);
+            expect(result.e).toEqual(current.e);
+        });
+    });
```
Comment (jstevenson) [inline packages/javascript-react/src/utilities/object-utils.test.ts:100]: Great work. A couple of use cases that could be add:

1. Comparing & replacing top-level arrays
2. Comparing & replacing top-level primitives (string, int, etc)

**Review by jstevenson — APPROVED** [review]
Looks great. I added some tests that could be included for simpler comparisons.

**Review by CodyFraker — APPROVED** [review]
Going to leave the relative filepaths alone but that'd be my only complaint. Would love a lesson on how to setup the tests for our services.

**Review by myty — COMMENTED** [review]
> Going to leave the relative filepaths alone but that'd be my only complaint. Would love a lesson on how to setup the tests for our services. 

yeah.  I'll create an issue for this.  would be nice to remove the relative paths.  it's currently a requirement.

**Pull request merged**